pkg/printers/json.go

```diff
@@ -9,22 +9,30 @@ import (
 	"github.com/golangci/golangci-lint/pkg/result"
 )
 
-type JSON struct{}
+type JSON struct {
+	silent bool
+}
 
-func NewJSON() *JSON {
-	return &JSON{}
+func NewJSON(silent bool) *JSON {
+	return &JSON{
+		silent: silent,
+	}
 }
 
 type JSONResult struct {
 	Issues []result.Issue
 }
 
-func (JSON) Print(ctx context.Context, issues <-chan result.Issue) (bool, error) {
+func (j JSON) Print(ctx context.Context, issues <-chan result.Issue) (bool, error) {
 	allIssues := []result.Issue{}
 	for i := range issues {
 		allIssues = append(allIssues, i)
 	}
 
+	if len(allIssues) == 0 && j.silent {
+		return false, nil
```

Comment:
As I [mentioned here](https://github.com/golangci/golangci-lint/issues/110#issuecomment-397825420), I'm not sure this is actually useful to do for JSON output. It's off by default so not a *huge* deal as such, but the "silent" flag is pretty generically named and could also change other behaviour in the future, and having it output invalid JSON as a kind of side-effect may surprise people.

It may also be confusing in some cases where users provide custom flags to the editor integration or some such, and the editor isn't prepared to deal with invalid JSON, causing confusing errors.

Personally I think always outputting valid JSON is much better, and would simplify the life of people implementing this (i.e. me :smile:).

Reply:
I agree about valid JSON, it would be better to not change JSON output at all

Reply:
Sure for JSON we can just emit {} perhaps or simply let it be as it was. @mmatczuk has a point though to consider.

What about the checkstyle xml? Perhaps it should also produce valid xml?

Tab and text can be silenced then is this the temporary consensus until a major decision is made as to the default behavior?

Reply:
Ah I just now saw you mentioned checkstyle as well @jirfag